pkg/ddl/ddl_worker.go

```diff
@@ -958,20 +958,22 @@ func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
 	err = w.registerMDLInfo(job, schemaVer)
 	if err != nil {
 		w.sess.Rollback()
-		d.unlockSchemaVersion(job.ID)
 		return 0, err
 	}
 	err = w.updateDDLJob(job, runJobErr != nil)
 	if err = w.handleUpdateJobError(t, job, err); err != nil {
 		w.sess.Rollback()
-		d.unlockSchemaVersion(job.ID)
 		return 0, err
 	}
 	writeBinlog(d.binlogCli, txn, job)
 	// reset the SQL digest to make topsql work right.
 	w.sess.GetSessionVars().StmtCtx.ResetSQLDigest(job.Query)
+	if !w.ddlCtx.isOwner() {
+		// This TiDB instance is not DDL owner anymore, it should not commit any transaction.
+		w.sess.Rollback()
+		return 0, dbterror.ErrNotOwner
+	}
```

Comment:
Should we do this before HandleJobDone？

Reply:
bc33faf467b5b875b9bc74e8163117c9e7c598b0